templates/tfengine/components/project/terraform.tfvars

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{hclField . "exists"}}
+project_id = "{{.project_id}}"
+parent_type = "{{.parent_type}}"
+parent_id = "{{.parent_id}}"
+billing_account = "{{.billing_account}}" 
+{{hclField . "apis"}}
+{{hclField . "is_shared_vpc_host"}}
+{{hclField . "api_identities"}}
+shared_vpc_attachment = {
+  {{- $host := get . "shared_vpc_attachment.host_project_id"}}
```

Comment:
You should use `get` with a default value, otherwise just access this field directly if you know that it has to exist.

Reply:
Didn't do it because it's required in the schema, so if that crashes because it wasn't provided, that's what is expected. Otherwise it won't crash if nothing is provided.

There are two options. as I see:

1. Adding a default here and removing the required from the schema (also setting the default there)
2. Leaving it like it is

Which one do you consider more correct?

Reply:
I'd prefer setting a default in the schema, that's the right place for it in my opinion. Here, if it's required, it should just be accessed, like `{{. shared_vpc_attachment.host_project_id}}`

Reply:
Okay, I did it with a "" as default.